contracts/libs/crypto/BN254.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+library BN254 {
+    uint256 internal constant SCALAR_FIELD_SIZE =
+        21888242871839275222246405745257275088548364400416034343698204186575808495617;
+    uint256 internal constant BASE_FIELD_SIZE =
+        21888242871839275222246405745257275088696311157297823662689037894645226208583;
+
+    struct Scalar {
+        uint256 data;
+    }
+
+    struct G1Affine {
+        uint256 x;
+        uint256 y;
+    }
+
+    function g1Basepoint() internal pure returns (G1Affine memory basepoint_) {
+        return G1Affine(1, 2);
+    }
+
+    function scalarFromUint256(uint256 a_) internal pure returns (Scalar memory scalar_) {
+        scalar_ = Scalar(a_);
+
+        require(validateScalar(scalar_), "BN: invalid scalar");
+    }
+
+    function scalarFromUint256ModScalar(uint256 a_) internal pure returns (Scalar memory scalar_) {
+        return Scalar(a_ % SCALAR_FIELD_SIZE);
+    }
+
+    function g1PointFromAffine(
+        uint256 x_,
+        uint256 y_
+    ) internal pure returns (G1Affine memory point_) {
+        point_ = G1Affine(x_, y_);
+
+        require(validateG1Point(point_), "BN254: invalid point");
+    }
+
+    function pointNeg(G1Affine memory a_) internal pure returns (G1Affine memory r_) {
```

Comment:
I think you can just remove "point" and "scalar" prefixes since parameter types are different.